File: examples/pointer_arithmetic/compound_assignments.c
// DECY-041: Test compound assignment operators
// This file tests +=, -=, *=, /=, %= operators

int increment_by(int value, int amount) {
    value += amount;
    return value;
}

int decrement_by(int value, int amount) {
    value -= amount;
    return value;
}

int multiply_by(int value, int factor) {
    value *= factor;
    return value;
}

int divide_by(int value, int divisor) {
    if (divisor != 0) {
        value /= divisor;
    }
    return value;
}

int modulo_by(int value, int modulus) {
    if (modulus != 0) {
        value %= modulus;
    }
    return value;
}

// Pointer arithmetic with compound assignments
void advance_pointer(int* ptr, int offset) {
    ptr += offset;
}

File: examples/pointer_arithmetic/compound_assignments.rs
static mut ERRNO: i32 = 0;
fn decrement_by(mut value: i32, mut amount: i32) -> i32 {
    value = value - amount;
    return value;
}
fn increment_by(mut value: i32, mut amount: i32) -> i32 {
    value = value + amount;
    return value;
}
fn divide_by(mut value: i32, mut divisor: i32) -> i32 {
    if divisor != 0 {
    value = value / divisor;
}
    return value;
}
fn modulo_by(mut value: i32, mut modulus: i32) -> i32 {
    if modulus != 0 {
    value = value % modulus;
}
    return value;
}
fn advance_pointer(mut ptr: *mut i32, mut offset: i32) {
    ptr = ptr.wrapping_add(offset as usize);
}
fn multiply_by(mut value: i32, mut factor: i32) -> i32 {
    value = value * factor;
    return value;
}
